File: Tests/CSharp/TestAllDatabases/TestAllDatabases/TestFixtures/Setup/SetupTests.cs
//======================================
// Copyright EntitySpaces, LLC 2005 - 2006 
//======================================

using System;
using System.Data;

using NUnit.Framework;
using BusinessObjects;
using EntitySpaces.Core;
using EntitySpaces.Interfaces;
using EntitySpaces.Profiler;

namespace Tests.Base
{
    [SetUpFixture]
	public class SetupTests
	{
        static object esEntity_ModifiedByHandler()
        {
            return "ModifiedBy Client User";
        }

        static object esEntity_AddedByEventHandler()
        {
            return "AddedBy Client User";
        }

        [SetUp]
        public void Init()
		{
            //esProviderFactory.Factory = new EntitySpaces.LoaderMT.esDataProviderFactory();
            esProviderFactory.Factory = new EntitySpaces.Loader.esDataProviderFactory();

            //ProfilerListener.BeginProfiling("EntitySpaces.SqlClientProvider",
            //    ProfilerListener.Channels.Channel_1);

            esEntity.AddedByEventHandler +=
                      new ModifiedByEventHandler(esEntity_AddedByEventHandler);

            esEntity.ModifiedByEventHandler +=
                      new ModifiedByEventHandler(esEntity_ModifiedByHandler);

            UnitTestBase.RefreshDatabase();
            UnitTestBase.RefreshForeignKeyTest();
        }
	}
}

File: Tests/VB/TestAllDatabases/TestAllDatabases/TestFixtures/Setup/SetupTests.vb
'======================================
' Copyright EntitySpaces, LLC 2005 - 2006 
'======================================

Imports System.Data

Imports NUnit.Framework
Imports BusinessObjects
Imports EntitySpaces.Interfaces

Namespace Tests.Base
	<SetUpFixture> _
	Public Class SetupTests
		<SetUp> _
		Public Sub Init()
            'esProviderFactory.Factory = New EntitySpaces.LoaderMT.esDataProviderFactory()
            esProviderFactory.Factory = New EntitySpaces.Loader.esDataProviderFactory()
			UnitTestBase.RefreshDatabase()
			UnitTestBase.RefreshForeignKeyTest()
		End Sub
	End Class
End Namespace
